proc_scripts/phasing.py

```diff
@@ -127,44 +131,49 @@ def ph1_real_Abs(s,dw):
         Resonance (San Diego, Calif. : 1997), 201(2), 230–238.
         https://doi.org/10.1016/j.jmr.2009.09.017
     '''
-    fl.push_marker() 
+    if fl is not None: fl.push_marker() 
     ph1 = nddata(r_[-5:5:70j]*dw,'phcorr')
-    dx = diff(ph1.getaxis('phcorr')[r_[0,1]]).item()
-    ph1 = exp(-1j*2*pi*ph1*s.fromaxis('t2'))
+    dx = np.diff(ph1.getaxis('phcorr')[r_[0,1]]).item()
+    ph1 = np.exp(-1j*2*pi*ph1*s.fromaxis('t2'))
     s_cost = s * ph1
     ph0 = s_cost.C.sum('t2')
     ph0 /= abs(ph0)
     s_cost /= ph0
-    fl.next('phasing cost function')
-    s_cost.run(real).run(abs).sum('t2')
-    fl.plot(s_cost,'.')
-    ph1_opt = s_cost.argmin('phcorr').item()
-    print('optimal phase correction',repr(ph1_opt))
+    if fl is not None:
+        fl.next('phasing cost function')
+    s_cost.run(np.real).run(abs).sum('t2')
+    s_cost = s_cost['ph2',1]['ph1',0]
+    if fl is not None:
+        fl.plot(s_cost,'.')
+    s_cost.smoosh(['vd','phcorr'],'transients')
+    ph1_opt = np.asarray(s_cost.argmin('transients').item())
+    logger.info(strm("THIS IS PH1_OPT",ph1_opt))
+    logger.info(strm('optimal phase correction',repr(ph1_opt)))
     # }}}
     # {{{ apply the phase corrections
+    #s_cost.chunk('transients',['vd','phcorr'],[30,-1])
     def applyphase(arg,ph1):
-        arg *= exp(-1j*2*pi*ph1*arg.fromaxis('t2'))
+        arg *= np.exp(-1j*2*pi*ph1*arg.fromaxis('t2'))
         ph0 = arg.C.sum('t2')
         ph0 /= abs(ph0)
         arg /= ph0
         return arg
     def costfun(ph1):
-        if type(ph1) is ndarray:
-            ph1 = ph1.item()
+        logger.info(strm("PH1 IS THIS:",ph1))
+        if type(ph1) is np.ndarray:
+            ph1 = ph1[0].item()
```

Comment:
164 is different in meaning than what it's replacing -- it means pull the first and use that -- it will not error out if there is more than one the way that the first statement will.

Reply:
yes, but the point should be that it should be a single number -- what you're describing should fail